lib/Varien/Data/Collection/Db.php

```diff
@@ -242,8 +242,14 @@ public function getSelectCountSql()
         $countSelect->reset(Zend_Db_Select::LIMIT_OFFSET);
         $countSelect->reset(Zend_Db_Select::COLUMNS);
 
-        $countSelect->columns('COUNT(*)');
-
+        if (count($this->getSelect()->getPart(Zend_Db_Select::GROUP)) > 0) {
+            $countSelect->reset(Zend_Db_Select::GROUP);
+            $countSelect->distinct(true);
+            $group = $this->getSelect()->getPart(Zend_Db_Select::GROUP);
+            $countSelect->columns("COUNT(DISTINCT " . implode(", ", $group) . ")");
```

Comment:
Untested...

```suggestion
            $group = array_map(function($token) {
                return $this->getSelect()->getAdapter()->quoteIdentifier($token, true);
            }, $group);
            $countSelect->columns("COUNT(DISTINCT " . implode(", ", $group) . ")");
```

Reply:
Ok, another approach. I will test this one.

Reply:
I've tested your suggestion and it works as expected. You never stop learning ;) Thank you @colinmollenhour